pandas/io/formats/style.py

```diff
@@ -802,6 +810,146 @@ def where(
             lambda val: value if cond(val) else other, subset=subset, **kwargs
         )
 
+    def set_tooltips(self, ttips: DataFrame):
+        """
+        Add string based tooltips that will appear in the `Styler` HTML result.
+
+        Parameters
+        ----------
+        ttips : DataFrame
+            DataFrame containing strings that will be translated to tooltips. Empty
```

Comment:
why ignore empty strings? (we don't normally treat these as nulls)

Reply:
Allowing empty string as valid will produce a tooltip on hover that doesn't give any additional information for the cell. It strikes me this will be more annoying than making sure it is removed - cant really imagine anyone wanting to create such a tooltip? However, happy to change if needed...